### PR TITLE
Add set of classes for CRUD documentation

### PR DESCRIPTION
**Pull request by Nek-**
This is a set of classes that will generate automatic documentation for CRUDs.

I'm not really happy with this implementation and I am open to suggestions.

Here is what it does: it adds a decorator to the basic documentation factory that looks in routing for CRUD controllers. For each CRUD controller, it adds a new documentation entry.

@Valouleloup @mcsky WDYT?

### CODE REVIEW

File: src/Bridge/Symfony/DependencyInjection/CompilerPass/DocumentationProviderCompilerPass.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Biig\Melodiia\Bridge\Symfony\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class DocumentationProviderCompilerPass implements CompilerPassInterface
```
Comment (Nek-) [inline src/Bridge/Symfony/DependencyInjection/CompilerPass/DocumentationProviderCompilerPass.php]: This class is useless and should be removed.

File: src/Documentation/OpenApiDocFactory.php [hunk 2]
```diff
@@ -15,13 +15,20 @@
...
     private $requestStack;
+
+    /** @var array */
     private $config;
 
-    public function __construct(RequestStack $requestStack, array $config)
+    /** @var DocuementationProviderInterface[] */
+    private $documentationProviders;
+
+    public function __construct(RequestStack $requestStack, array $config, array $documentationProviders = [])
```
Comment (Nek-) [inline src/Documentation/OpenApiDocFactory.php]: Oops. $documentationProviders to be remove.

File: src/Bridge/Symfony/Routing/CrudDocumentationFactory.php [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        $tag = $model;
+
+        if (isset($defaults[self::DOCUMENTATION_TAG])) {
+            $tag = $defaults[self::DOCUMENTATION_TAG];
+        }
+
+        $annot = new Post([
+            'path' => $this->cleanPath($route->getPath()),
+            'responses' => [
+                new Response([
```
Comment (mcsky) [inline src/Bridge/Symfony/Routing/CrudDocumentationFactory.php:129]: The API should return a 400 if request don't match all the constraints.

Please see [linked issue ](https://github.com/biig-io/Melodiia/issues/7)

**Review by mcsky — COMMENTED** [review]
Seem good to me, but I'm not very used to with `CompilerPass`

**Pull request merged**